src/transports/HTTPTransport.ts

```diff
@@ -27,8 +29,19 @@ class HTTPTransport implements ITransport {
       this.onDataCallbacks.map((cb) => {
         cb(result);
       });
+    }).catch(async (e) => {
```

Comment:
we might just use async await all the way down so we have less .then promise catches. Might look a little cleaner